### PR TITLE
XML Sitemap: Incorrect encoding of special characters

### PR DESCRIPTION
**Pull request by contactashish13**
https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/2110

### CODE REVIEW

**Comment by wpsmort** [comment]
@contactashish13 This works as expected.  I only came across one problem as follows:

When a less than symbol is entered in the Title or Caption fields in the Edit Image dialog, the sitemap is displayed as a blank page.

Steps to reproduce this:
1. Create a post and set a featured image
2. Click on the featured image to open the popup dialog where you can set the title and caption
3. Enter < 2 into the Caption field and update the post
4. Open the sitemap and it should be a blank page.

**Comment by contactashish13** [comment]
@wpsmort have fixed for this as well.

**Comment by wpsmort** [comment]
This is tested and is good.

File: inc/aiosp_common.php [hunk 1]
```diff
@@ -174,12 +174,26 @@ public static function is_url_valid( $url ) {
...
+		}
+
+		// sanitize the other tags.
 		if ( in_array( $tag, array( 'guid', 'link', 'loc', 'image:loc' ) ) ) {
 			$value = esc_url( $value );
-		} else if( ! is_array( $value ) ) {
-			$value = htmlspecialchars( $value, ENT_QUOTES );
+		} else {
+			// some tags contain sanitized to some extent but they do not encode < and >.
+			if ( ! in_array( $tag, array( 'image:title' ) ) ) {
```
Comment (EkoJR) [inline inc/aiosp_common.php]: Missing 3rd param boolean set as true for strict operations

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -3,9 +3,13 @@ name: Bug report
 about: Create a report to help us improve
 
 ---
+ - [ ] I understand that this is a development repository and not a support forum. This is a bug report and is not a support request.
```
Comment (EkoJR) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Why is this file being modified?

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ## Proposed changes
 
-Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
```
Comment (EkoJR) [inline PULL_REQUEST_TEMPLATE.md]: Why is this file being modified?

File: all_in_one_seo_pack.php [hunk 4]
```diff
@@ -4,7 +4,7 @@
 Plugin Name: All In One SEO Pack
 Plugin URI: https://semperplugins.com/all-in-one-seo-pack-pro-version/
 Description: Out-of-the-box SEO for your WordPress blog. Features like XML Sitemaps, SEO for custom post types, SEO for blogs or business sites, SEO for ecommerce sites, and much more. More than 30 million downloads since 2007.
-Version: 2.10
+Version: 2.10.1
```
Comment (EkoJR) [inline all_in_one_seo_pack.php]: Why is this file being modified?

This should be left for the Version Change PR. It is also outdated.

File: readme.txt [hunk 5]
```diff
@@ -4,7 +4,7 @@ Donate link: https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=mrtor
 Tags: SEO, all in one seo, WordPress SEO, Google Search Console, XML Sitemap, image seo
 Requires at least: 4.4
 Tested up to: 5.0
-Stable tag: 2.10
```
Comment (EkoJR) [inline readme.txt]: Why is this file being modified?

This should be left for the Version Change PR. It is also outdated.

File: inc/aiosp_common.php (hunk 1 above)
Comment (EkoJR) [reply inc/aiosp_common.php]: http://php.net/manual/en/function.in-array.php

**Comment by contactashish13** [comment]
@EkoJR the files you have tagged are part of the merge. They have not been changed by me, if you check my commits.

**Comment by EkoJR** [comment]
@contactashish13 Unfortunately, it will still affect 2.11 branch when it is merged.

**Comment by EkoJR** [comment]
@contactashish13 @michaeltorbert I see what happened. This branch was created off of master, instead of 2.11, and master contains changes that 2.11 doesn't have (in a commit sense).

I would recommend rebasing this on top of 2.11 branch, or merging into master, but I think rebasing to 2.11 would be better.

**Comment by EkoJR** [comment]
Which I already have a branch for https://github.com/EkoJR/all-in-one-seo-pack/tree/i-2110

**Pull request closed without merging**